app/scripts/controllers/metametrics-controller.ts

```diff
@@ -584,7 +584,6 @@ export default class MetaMetricsController extends BaseController<
       : {};
 
     this.update((state) => {
-      // @ts-expect-error this is caused by a bug in Immer, not being able to handle recursive types like Json
```

Comment:
Hmm. Interesting. Any idea why this error went away?

Reply:
Not clear why this happened but I added the same one here too https://github.com/MetaMask/metamask-extension/pull/28784/files#diff-037120a57cb346b20107f8eea73f9fc79a23738050db9828196e0eefdd8ffc00R428